ee/presencedetection/presencedetection.go

```diff
@@ -0,0 +1,51 @@
+package presencedetection
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+const DetectionFailedDurationValue = -1 * time.Second
+
+type PresenceDetector struct {
+	lastDetectionUTC          time.Time
+	mutext                    sync.Mutex
+	hadOneSuccessfulDetection bool
```

Comment:
You probably don't need the bool, but it's reads cleaner than checking if `lastDetectionUTC` is 0

Reply:
agree, even checking for zero is not necessary becaues we are saying if `time.Since(0) < interval` ... as long as the interval is not greater than the begging of time =)

Reply:
welp, after I wrote that out .... i thought, what if we added an interval that was a century ... and put it back